server/txSender/txSender.go

```diff
@@ -0,0 +1,195 @@
+package txSender
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	coreTx "github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-sdk-go/core"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+const (
+	waitTimeRetrialsMs = 50
+)
+
+// TxSenderArgs holds args to create a new tx sender
+type TxSenderArgs struct {
+	Wallet                core.CryptoComponentsHolder
+	Proxy                 Proxy
+	TxInteractor          TxInteractor
+	DataFormatter         DataFormatter
+	SCBridgeAddress       string
+	MaxRetrialsGetAccount int
+}
+
+type txSender struct {
+	wallet                core.CryptoComponentsHolder
+	proxy                 Proxy
+	netConfigs            *data.NetworkConfig
+	txInteractor          TxInteractor
```

Comment:
wrong approach should use the nonceTransactionsHandler struct that is able to handle the nonce internally & also handles the resending mechanism. Was done by implementing the best practices coined by our teams. https://github.com/multiversx/mx-sdk-go/blob/main/interactors/nonceHandlerV2/nonceTransactionsHandler.go

Reply:
Good suggestion, used the core component for nonce management instead